test/Product_test.jl

```diff
@@ -1,4 +1,4 @@
-@testset_skip "Product ansatz" begin
+@testset "Product ansatz" begin
```

Comment:
Add a test for `Product` with `arrays::AbstractArray{<:AbstractMatrix}`.

Reply:
we already have